app/gateway/api/handler/delivery_handler.go

```diff
@@ -63,3 +67,48 @@ func (h *Handler) ListDeliveries() http.HandlerFunc {
 		rest.SendJSON(rw, resp.Status, resp.Payload, resp.Headers) //nolint:errcheck
 	}
 }
+
+func (h *Handler) CreateDelivery() http.HandlerFunc {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		deliveryBody := &schema.CreateDeliveryRequest{}
+		err := json.NewDecoder(req.Body).Decode(deliveryBody)
+
+		var resp *response.Response
+
+		if err != nil {
+			resp = response.InternalServerError(err)
+			rest.SendJSON(rw, resp.Status, resp.Payload, resp.Headers) //nolint:errcheck
+
+			return
+		}
+
+		err = schema.ValidateCreateDeliveryRequest(deliveryBody)
+		if err != nil {
+			resp := response.BadRequest(err, err.Error())
+			rest.SendJSON(rw, resp.Status, resp.Payload, resp.Headers) //nolint:errcheck
+
+			return
+		}
+
+		useCaseInput := usecase.CreateDeliveryInput{}
+		useCaseInput.Delivery.Qty = deliveryBody.Qty
+		useCaseInput.Delivery.Reference = strconv.Itoa(rand.Intn(schema.MaxReference) + schema.MinReference) //nolint:gosec
```

Comment:
Fazer um get pelo reference para entender se ele já existe. Caso já exista, tenta criar de novi. Utilizar um for com break para essa lógica

Reply:
Feito, quando puder avaliar de novo. Aproveitei e já adicionei a rota de buscar por referência